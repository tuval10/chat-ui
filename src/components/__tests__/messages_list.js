import React from 'react';
import {create} from 'react-test-renderer';
import MessagesList from '../messages_list';
import {user as myUser, message, otherUserMessage} from '../../test_helpers/fixtures'

describe('MessagesList component', () => {
  describe('renders correctly', () => {
    // noinspection ES6ModulesDependencies
    it('when there are messages', () => {
      const tree = create(
        <MessagesList {...{myUser, messages: [message, otherUserMessage]}} />
      ).toJSON();
      expect(tree).toMatchSnapshot();
    });

    it('when there are no messages', () => {
      const tree = create(
        <MessagesList {...{myUser, messages: []}} />
      ).toJSON();
      expect(tree).toMatchSnapshot();
    });
  });
});