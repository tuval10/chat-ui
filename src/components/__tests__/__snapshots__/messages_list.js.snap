// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MessagesList component renders correctly when there are messages 1`] = `
<div
  className="messages-list"
>
  <div
    className="message-item mine"
  >
    <div
      className="message-details"
    >
      <div
        className="avatar-container"
      >
        <img
          alt="user-logo"
          className="avatar-icon"
          src="005-bullbasaur.png"
        />
      </div>
      <div
        className="username"
      >
        stub_username
      </div>
      <div
        className="posted-at"
      >
        00:34
      </div>
    </div>
    <div
      className="message-text"
    >
      stub message
    </div>
  </div>
  <div
    className="message-item"
  >
    <div
      className="message-details"
    >
      <div
        className="avatar-container"
      >
        <img
          alt="user-logo"
          className="avatar-icon"
          src="004-jigglypuff.png"
        />
      </div>
      <div
        className="username"
      >
        stub_username 2
      </div>
      <div
        className="posted-at"
      >
        00:35
      </div>
    </div>
    <div
      className="message-text"
    >
      stub message
    </div>
  </div>
</div>
`;

exports[`MessagesList component renders correctly when there are no messages 1`] = `
<div
  className="messages-list"
/>
`;
